src/connector/src/connector_common/iceberg/mod.rs

```diff
@@ -442,20 +476,35 @@ impl IcebergCommon {
     ) -> ConnectorResult<Table> {
         match self.catalog_type() {
             "storage" => {
-                let config =
-                    storage_catalog::StorageCatalogConfig::builder()
-                        .warehouse(self.warehouse_path.clone().ok_or_else(|| {
-                            anyhow!("`warehouse.path` must be set in storage catalog")
-                        })?)
-                        .access_key(self.access_key.clone().ok_or_else(|| {
-                            anyhow!("`s3.access.key` must be set in storage catalog")
-                        })?)
-                        .secret_key(self.secret_key.clone().ok_or_else(|| {
-                            anyhow!("`s3.secret.key` must be set in storage catalog")
-                        })?)
-                        .region(self.region.clone())
-                        .endpoint(self.endpoint.clone())
-                        .build();
+                // check gcs credential or s3 access key and secret key
+                let config = if self.gcs_credential.is_none() {
```

Comment:
Is it possible to identify the storage type according to the URL schema (`gcs://...`, `s3://...`, etc.)? It seems more elegant and friendly for adding Azure support in the future.